client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx

```diff
@@ -188,6 +218,19 @@ export const AnalysisWizard: React.FC<IAnalysisWizard> = ({
     mode: "all",
   });
 
+  useEffect(() => {
```

Comment:
It seems to me that applications cannot change once the wizard was opened.
Can you describe the scenario that requires re-setting the defaults?

Reply:
The bug occurred because the call to the function was performed in a click event, and in the case of a source type application, the value of the variable was not updated during the call to the function. As a result, only the binary files were transferred successfully.
To fix the bug, we added the useEffect function to ensure that the variable is updated automatically whenever the component is updated according to changes in the applications data. This way the useEffect ensures that the default value will be true and avoids a situation where an incorrect state is first displayed after the component is loaded.

Reply:
issue #1837

Reply:
FYI I need to check this case.

Reply:
This `useEffect()` does not make much sense.  It reads as "Use the default mode unless the default mode is falsey. If it is falsey, set a default of `source-code-deps`".

Really this could be done up in the `useForm` block.

Reply:
I fixed it